### PR TITLE
add mysql support

### PR DESCRIPTION
**Pull request by hooopo**
Add MySQL support.

### CODE REVIEW

**Comment by eviltrout** [comment]
Looks very cool to me, but I will delegate to @SamSaffron here as it's a big change.

**Comment by SamSaffron** [comment]
This looks amazing!!!! Can you update the benchmarks in the benchmark directory? Curious to see how this stacks up?

**Comment by hooopo** [comment]
@SamSaffron updated.

File: bench/topic_mysql_perf.rb [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+  end
+  r.compare!
+end
+
+# Comparison:
+# mysql select title id:      486.5 i/s
+# mini_sql query_single title id:      450.3 i/s - same-ish: difference falls within error
+# sequel title id pluck:      367.9 i/s - 1.32x  slower
+# sequel title id select:      352.5 i/s - 1.38x  slower
+# mini_sql select title id:      346.2 i/s - 1.41x  slower
```
Comment (SamSaffron) [inline bench/topic_mysql_perf.rb]: I am a bit concerned about this number is there anything we can do to improve this?

Comment (hooopo) [reply bench/topic_mysql_perf.rb]: improved 😃

**Comment by SamSaffron** [comment]
excellent work! merging it in, will look at doing a new release this week

**Pull request merged**